src/nodelets/icp_odometry.cpp

```diff
@@ -84,6 +88,24 @@ class ICPOdometry : public rtabmap_ros::OdometryROS
 		pnh.param("scan_range_max",         scanRangeMax_, scanRangeMax_);
 		pnh.param("scan_voxel_size",        scanVoxelSize_, scanVoxelSize_);
 		pnh.param("scan_normal_k",          scanNormalK_, scanNormalK_);
+
+		if (pnh.hasParam("plugins"))
+		{
+			XmlRpc::XmlRpcValue pluginsList;
+	    	pnh.getParam("plugins", pluginsList);
+
+		    for (int32_t i = 0; i < pluginsList.size(); ++i)
+		    {
+		      std::string pluginName = static_cast<std::string>(pluginsList[i]["name"]);
+		      std::string type = static_cast<std::string>(pluginsList[i]["type"]);
+		      NODELET_INFO("%s: Using plugin \"%s\"", pluginName.c_str(), type.c_str());
+
+		      boost::shared_ptr<rtabmap_ros::PluginInterface> plugin = plugin_loader_.createInstance(type);
```

Comment:
What happens when the plugin doesn't exist? Is it returning a null pointer or there is an exception?

Reply:
I added a try catch, since createInstance can throw PluginlibExceptions.